src/Microsoft.ML.TorchSharp/Utils/ImageUtils.cs

```diff
@@ -0,0 +1,221 @@
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using static TorchSharp.torch.utils.data;
+using static TorchSharp.torch;
+using TorchSharp;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.TorchSharp.Utils
+{
+    internal class ImageUtils
+    {
+
+        public static void Postprocess(Tensor imgBatch, Tensor classification, Tensor regression, Tensor anchors, out UInt32[] predictedLabels, out Single[] score, out Single[] boxes, double scoreThreshold = 0.05, double overlapThreshold = 0.5)
+        {
+            predictedLabels = null;
+            score = null;
+            boxes = null;
+
+            if (imgBatch is null)
+            {
+                throw new ArgumentNullException(nameof(imgBatch));
+            }
+
+            if (classification is null)
+            {
+                throw new ArgumentNullException(nameof(classification));
+            }
+
+            if (regression is null)
+            {
+                throw new ArgumentNullException(nameof(regression));
+            }
+
+            if (anchors is null)
+            {
+                throw new ArgumentNullException(nameof(anchors));
+            }
+
+            using (var postprocessScope = torch.NewDisposeScope())
+            {
+                var transformedAnchors1 = TransformBbox(anchors, regression);
+                var transformedAnchors = ClipBoxes(transformedAnchors1, imgBatch);
+
+                var finalResult = new[] { new List<Tensor>(), new List<Tensor>(), new List<Tensor>() };
+
+                for (int i = 0; i < classification.shape[2]; ++i)
+                {
+                    var scores1 = torch.squeeze(classification[.., .., i]);
+                    var scoresOverThresh = scores1 > 0.05;
+                    if (scoresOverThresh.sum().ToSingle() == 0)
+                    {
+                        // no boxes to NMS, just continue
+                        continue;
+                    }
+
+                    var scores = scores1[scoresOverThresh];
+                    var anchorBoxes1 = torch.squeeze(transformedAnchors);
+                    var anchorBoxes = anchorBoxes1[scoresOverThresh];
+                    var anchorsNmsIdx = Nms(anchorBoxes, scores, overlapThreshold);
+                    var finalAnchorBoxesIndexesValue = torch.ones(anchorsNmsIdx.shape[0], dtype: ScalarType.Int64, device: imgBatch.device).multiply(i);
+
+                    finalResult[0].Add(scores[anchorsNmsIdx]);
+                    finalResult[1].Add(finalAnchorBoxesIndexesValue);
+                    finalResult[2].Add(anchorBoxes[anchorsNmsIdx]);
+                }
+
+                int boxIndex = 0;
+
+                if (finalResult[0].Count > 0)
+                {
+                    var finalScores = torch.cat(finalResult[0], dim: 0);
+                    var finalAnchorBoxesIndexes = torch.cat(finalResult[1], dim: 0);
+                    var finalAnchorBoxesCoordinates = torch.cat(finalResult[2], dim: 0);
```

Comment:
Do you think we need to round anchor boxes coordinate to integer considering that we are taking the 
absolute coordinates of bounding boxes as input?

Reply:
Well, the input we are taking for bounding boxes is a float, so in theory someone could pass in unrounded values if they wanted to. I would leave this as is and if we have feedback/complaints about it we can revisit it.